src/mesos_state_client.erl

```diff
@@ -51,13 +102,7 @@ poll(URI) ->
   {ok, Hostname} = inet:gethostname(),
   UserAgent = lists:flatten(io_lib:format("Mesos-State / Host: ~s, Pid: ~s", [Hostname, os:getpid()])),
   Headers = [{"Accept", "application/json"}, {"User-Agent", UserAgent}],
-  Headers1 =
-    case application:get_env(?APP, token) of
-      undefined ->
-        Headers;
-      {ok, Value} ->
-        [{"Authorization", Value}|Headers]
-    end,
+  Headers1 = maybe_use_token(Headers),
```

Comment:
I would actually do something like:
```
Authorization = application:get_env(?APP, token, <<"anonymous">>),
Headers [{"Accept", "application/json"}, {"User-Agent", UserAgent}, {"Authorization", Authorization}]
```

And then have the code in mesos_state_app to do the loading of the token. Right now, we don't have one, but add a file that implements the application behaviour (http://erlang.org/doc/man/application.html), and load it there. You'll also need to set mod in app.src.